core/trino-main/src/main/java/io/trino/execution/RenameTableTask.java

```diff
@@ -56,6 +58,30 @@ public String getName()
     {
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getSource());
+
+        // those two guards are here because some connectors violate getTableHandle contract for views or materialized views.
```

Comment:
nit: 
```suggestion
        // those two guards are here because some connectors violate SPI contract and return views or materialized views from getTableHandle.
```